src/main/java/core/basesyntax/supplier/BasicRandomValueSupplier.java

```diff
@@ -0,0 +1,8 @@
+package core.basesyntax.supplier;
+
+import core.basesyntax.behaviour.Suppliable;
+import java.util.Random;
+
+public abstract class BasicRandomValueSupplier implements Suppliable {
```

Comment:
Since BasicRandomValueSupplier is an abstract class with only one field, consider using an interface with a default method instead to provide the Random instance.